docs_src/microservices/security/Ch-SSH-Tunneling-HowToSecureDeviceServices.md

```diff
@@ -0,0 +1,297 @@
+
+### SSH Tunneling: Setup the SSH server on the remote machine
+
+Use the example from Docker hub, an example of ssh damenon or ssh server can be setup pretty easily as a Docker container: https://docs.docker.com/engine/examples/running_ssh_service/
+
+Note that this one is the ssh server and it is set up using password authentication by default.  In order to authenticate to this ssh server without password prompt, we injected the generated public SSH key from the primary machine via simple login into the ssh server machine first and then created the `authorized_keys` under `~/.ssh` directory.  Without loss of generality, the following command example shows how this is accomplished:
+
+```sh
+root@sshd-remote: mkdir -p ~/.ssh
+root@sshd-remote: chmod 700 ~/.ssh
+root@sshd-remote:  echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDKvsFf5HocBOBWXdVJKfQzkhf0K8lSLjZn9PX84VdhHyP8n1mzfpZywA4vsz8+A3OsGHAr2xpkyzOS0YkwD7nrI3q1x0A0+ANhQNOaKbnfQRe... root" >> ~/.ssh/authorized_keys
+```
+
+The ssh key pairs can be generated using `ssh-keygen` command from the primary machine and the contents of ssh public key usually is stored as ~/.ssh/id_rsa.pub file like this:
+
+```sh
+ssh-keygen -q -t rsa -C root -N '' -f ~/.ssh/id_rsa 2>/dev/null
+```
+
+One example of build script to make SSH public key be injected is given as like:
+
+[example build shell script to build and install SSH public key into sshd server](https://github.com/edgexfoundry-holding/external-device-security-examples/ssh-tunneling/build.sh)
+
+### SSH Tunneling: Local Port Forwarding
+
+from primary to secondary /remote
+it is achieved by -L flags of ssh command.
+
+```sh
+ssh -vv -o StrictHostKeyChecking=no \
+  -o UserKnownHostsFile=/dev/null \
+  -N $TUNNEL_HOST \
+  -L *:$SERVICE_PORT:$SERVICE_HOST:$SERVICE_PORT \
+  -p $TUNNEL_SSH_PROT
+```
+
+where environment variables are:
+
+- TUNNEL_HOST is the remote host name or IP address that SSH daemon or server is running on;
+
+- TUNNEL_SSH_PROT is the port number to be used on the SSH tunnel communication between the primary machine and the remote machine
+
+- SERVICE_PORT is the port number from the local or the primary to be forwared to the remote machine; without lose of generality, the port number on the remote machine is the same as the primary one
+
+- SERVICE_HOST is the service host name or IP address of the Docker containers that are running on the remote machine;
+
+### SSH Reverse Tunneling: Remote Port Forwarding
+
+the reverse direction: from the secondary /remote back to the primary
+
+The reverse SSH tunneling is also needed because the device services depends on the core services like `data`, `metadata`, ... and so on.  These core services are running on the primray machine and should be **reversely** tunneling back to the device services on the remote side through the SSH remote port forwarding connection.  This can be achieved by using `-R` flag of ssh command.
```

Comment:
Be specific here, list out all the services that the device services depend on. "... and so on" doesn't provide clarity.

primray => local
 **reversely** -> **reverse**